fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/postgresql/PostgreSqlResourceDAO.java

```diff
@@ -261,14 +252,11 @@ public Integer readResourceTypeId(String resourceType) throws FHIRPersistenceDBC
                 logger.finer("DB read resource type id complete. executionTime=" + dbCallDuration + "ms");
             }
             resourceTypeId = stmt.getInt(2);
-        } catch(FHIRPersistenceDBConnectException e) {
-            throw e;
```

Comment:
interesting... so this was never used?

Reply:
The DAO is passed a valid connection so it never attempts to connect itself, and therefore will never see FHIRPersistenceDBConnectException. This is a lot cleaner, because connection issues are handled higher up, leaving the DAO to focus on performing statements.